modules/promotion/src/Entity/Promotion.php

```diff
@@ -243,9 +243,23 @@ public function apply(EntityInterface $entity) {
     $entity_type_id = $entity->getEntityTypeId();
     // @todo should whatever invokes this method be providing the context?
     $context = new Context(new ContextDefinition('entity:' . $entity_type_id), $entity);
+
+    // Execute each plugin, this is an AND operation.
+    // @todo support OR operations.
+    /** @var \Drupal\commerce\Plugin\Field\FieldType\PluginItem $item */
+    foreach ($this->get('conditions') as $item) {
```

Comment:
Whats the point of splitting it up though then? aren't we just allowing an unnecessarily dangerous thing?